services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/GenericMultipartHelper.java

```diff
@@ -59,7 +60,8 @@ public void handleException(CompletableFuture<ResponseT> returnFuture,
                                 Throwable throwable) {
         Throwable cause = throwable instanceof CompletionException ? throwable.getCause() : throwable;
 
-        if (cause instanceof Error) {
+        if (cause instanceof Error || cause instanceof SdkException) {
+            cause.addSuppressed(SdkClientException.create(message.get()));
```

Comment:
Did you check if `initCause` could be used in this instance, and what the difference is (if any)?

Reply:
initCause would just add this new exception as the cause, which seems to defeat the purpose of this PR. I followed joinLikeSync approach https://github.com/aws/aws-sdk-java-v2/blob/feature/master/sra-identity-auth/utils/src/main/java/software/amazon/awssdk/utils/CompletableFutureUtils.java#L254

Reply:
I wasn't sure if it would overwrite, but yes I agree.